digiwf-email-integration/src/main/java/io/muenchendigital/digiwf/email/integration/domain/service/MailingService.java

```diff
@@ -62,20 +63,27 @@ public void sendMail(final Mail mail) throws MissingInformationMailException {
             helper.setText(mail.getBody());
             helper.setFrom(fromAdress);
 
-            if (mail.hasReplyTo()) {
+            if (StringUtils.isNotBlank(mail.getReplyTo())) {
                 helper.setReplyTo(mail.getReplyTo());
             }
 
-            if (mail.hasAttachement()) {
-                for (val attachmentPath : mail.getAttachmentPaths()) {
-                    final byte[] binaryFile = this.documentStorageFileRepository.getFile(
-                            attachmentPath,
-                            3
-                    );
-                    final Tika tika = new Tika();
-                    val file = new ByteArrayDataSource(binaryFile, tika.detect(binaryFile));
-                    val fileName = StringUtils.substringAfterLast(attachmentPath, "/");
-                    helper.addAttachment(fileName, file);
+            if (mail.getAttachments() != null && mail.getAttachments().size() > 0) {
+                for (val attachment : mail.getAttachments()) {
+                    if (StringUtils.isNotBlank(attachment.getAttachmentPath()) && StringUtils.isNotBlank(attachment.getDocumentStorageUrl())) {
```

Comment:
put validationlogic in a method.

e.g. "isAttachmentPathAndDocumentStorageNotBlank"

Reply:
done